src/components/app/details/triggerView/CIMaterialModal.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { Progressing, VisibleModal, stopPropagation, usePrompt } from '@devtron-labs/devtron-fe-common-lib'
+import CIMaterial from './ciMaterial'
+import { ReactComponent as CloseIcon } from '../../../../assets/icons/ic-close.svg'
+import { CIMaterialModalProps, CIMaterialRouterProps } from './types'
+
+export const CIMaterialModal = ({
+    loader,
+    closeCIModal,
+    isLoading,
+    abortController,
+    ...props
+}: CIMaterialModalProps) => {
+    const { ciNodeId } = useParams<Pick<CIMaterialRouterProps, 'ciNodeId'>>()
+    usePrompt({ shouldPrompt: isLoading })
+
+    useEffect(() => {
+        return () => {
+            abortController.abort()
```

Comment:
Please use abortPreviousRequest here.

Reply:
controller used here is used from props which comes from a class based component, cannot pass ref in abortPreviousReq